blockchain/chain.py

```diff
@@ -39,23 +39,23 @@ def calculateHash(self):
             Method to calculate hash from metadata
         """
         hashData = str(self.index) + str(self.data) + self.timestamp + self.previousHash + str(self.nonce)
-        return hashlib.sha256(hashData).hexdigest()
+        return hashlib.sha256(bytes(ord(x) for x in hashData)).hexdigest()
 
     def mineBlock(self, difficulty):
         """
             Method for Proof of Work
         """
-        print Back.RED + "\n[Status] Mining block (" + str(self.index) + ") with PoW ..."
+        print(Back.RED + "\n[Status] Mining block (" + str(self.index) + ") with PoW ...")
```

Comment:
probably use fstrings. Since this is cli we should use click and show messages with click.echo

Reply:
I opted for minimal change. If we are ok about putting 3.6 as lower bound why not.

Reply:
but I agree this doesn't block your PR